### ISSUE TITLE
numpy's int64 is automatically converted to float

### ISSUE THREAD

**Issue by Pastafarianist** [sympy/sympy#11086]
``` python
import numpy as np
import sympy as sp

x = sp.symbols('x')
print(x * 1)
print(x * np.int64(1))
```

Output:

```
x
1.0*x
```

I think it is unexpected that `int64` is cast to `float`. Perhaps it should be cast to `int`?

**Comment by isuruf** [comment]
This seems to be an issue on 32 bit systems where `np.int64(1)` is not a child of `int`.
Note that `np.int32(1)` is converted to a sympy integer because `np.int32` is a child class of `int`

**Comment by Pastafarianist** [comment]
I am running a 64-bit system:

    $ uname -srm                                                                                                                                                      
    Linux 4.1.15-gentoo-r1 x86_64

**Comment by isuruf** [comment]
What's the result of,
```
import numpy as np
from inspect import getmro
n = np.int64(1)
print(getmro(n.__class__))
```

**Comment by Pastafarianist** [comment]
```python
$ python
Python 3.4.3 (default, Jan  7 2016, 17:38:12) 
[GCC 4.8.5] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> from inspect import getmro
>>> n = np.int64(1)
>>> print(getmro(n.__class__))
(<class 'numpy.int64'>, <class 'numpy.signedinteger'>, <class 'numpy.integer'>, <class 'numpy.number'>, <class 'numpy.generic'>, <class 'object'>)
```

**Comment by asmeurer** [comment]
I guess we should add sympify translators for numpy types (if it is installed). This should be done in a way that doesn't import numpy unless it is explicitly used.

**Comment by rajeshkumargp** [comment]
Hi Could I work on this issue ?

**Comment by BrickMoon** [comment]
I take it this issue has yet to be resolved. I suppose I will take a stab at it, then.

**Comment by ArifAhmed1995** [comment]
Maybe this issue has been fixed by some other merged pull request.

```
>>> import sympy as sp
>>> import numpy as np
>>> x = sp.symbols('x')
>>> print(x * 1)
x
>>> print(x * np.int64(1))
x
>>>

```
I am also using a 64-bit system : 
```
$ uname -srm
Linux 3.19.0-15-generic x86_64
```

**Comment by isuruf** [comment]
@ArifAhmed1995, are you using Python 3?

**Comment by ArifAhmed1995** [comment]
@isuruf Oops ! Should have mentioned it. No , I am using `Python 2.7.12`

**Comment by asmeurer** [comment]
I see the issue in 64-bit Python 3:

```
In [23]: sympify(np.int64(1))
Out[23]: 1.00000000000000
```

**Comment by luukko** [comment]
I got really burned by this recently:
```
import numpy
import sympy
from sympy.physics.hydrogen import R_nl

def radial_integral(n, l, np, lp):
    r = sympy.Symbol("r", positive=True)
    integral = sympy.integrate(R_nl(np,lp,r)*R_nl(n,l,r)*r**2, (r, 0, sympy.oo))
    return float(integral)

int_args = [30, 3, 30, 4]
int64_args = [ numpy.int64(a) for a in int_args ]

print(radial_integral(*int_args))
print(radial_integral(*int64_args))
```
Output:
```
-0.9910712498212337
8913964512.0
```
I just thought I'll share this to point out that this bug has serious consequences in formulas that simplify nicely with integers but not with floats.
I'm also using 64-bit Python 3.

**Comment by ArifAhmed1995** [comment]
#11666 fixes this . Hope it will be merged soon.

```
-0.991071249821
-0.991071249821

```

**Comment by luukko** [comment]
Let's hope so. This was a nasty bug to track down because of course everything works in unit tests with actual integer constants... Bugs like this make me think maybe duck typing is not so great ater all :wink:. Thanks for the pull request @ArifAhmed1995!

**Comment by cbm755** [comment]
I think this was fixed by #11666 but not marked closed.  I did test it but please do reopen if I'm wrong.
